moto/rds/models.py

```diff
@@ -4508,8 +4513,11 @@ def _create_green_instance(
     ) -> str:
         source_instance: DBInstance | DBCluster = self._get_instance(self.source)
         green_instance: DBInstance | DBCluster
-        if source_instance.manage_master_user_password:
-            raise SourceDatabaseNotSupportedFault(source_instance.arn)
+        if hasattr(source_instance, "master_user_secret"):
+            if isinstance(source_instance, DBInstance):
+                raise SourceDatabaseNotSupportedFault(source_instance.arn)
+            else:
+                raise SourceClusterNotSupportedFault(source_instance.arn)
```

Comment:
Would you mind extracting this into a helper method, something like `valid_bgd_source_or_raise`?